src/pages/Popup/components/timer.jsx

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from "react";
+class Timer extends Component {
+    today = new Date();
+    constructor(props) {
+        super(props);
+        this.state = {
+            remTime: this.today
+        }
+    }
+
+    async componentDidMount() {
+        let alarms = await new Promise((resolve) => chrome.alarms.getAll(resolve));
+        alarms.sort(function (a, b) {
+            return a.scheduledTime - b.scheduledTime;
+        });
+        let timeobj = alarms[0];
+        let time = timeobj.scheduledTime;
+        console.log(typeof(alarms));
+        console.log(typeof(timeobj));
+        this.setState(() => ({
+            remTime: time - this.today
+        }))
+        this.myInterval = setInterval(() => {
+            const { remTime } = this.state;
+            if (remTime > 0) {
+                this.setState(({ remTime }) => ({
+                    remTime: remTime - 1000
+                }))
+            }
+            if (remTime <= 0) {
+                let alarms = new Promise((resolve) => chrome.alarms.getAll(resolve));
```

Comment:
@Shantanu-Bakshi I think the problem is here
This will have to be an async call! Make the corresponding function asynchronous
This is the reason why it is showing undefined property because the function is not waiting for data to load